include/rocksdb/options.h

```diff
@@ -303,8 +303,23 @@ struct ColumnFamilyOptions : public AdvancedColumnFamilyOptions {
 
 enum class WALRecoveryMode : char {
   // Original levelDB recovery
-  // We tolerate incomplete record in trailing data on all logs
-  // Use case : This is legacy behavior
+  //
+  // We tolerate the last record in any log to be incomplete due to a crash
+  // while writing it. Zeroed bytes from preallocation are also tolerated in the
+  // trailing data of any log.
+  //
+  // Use case: Applications for which updates, once applied, must not be rolled
+  // back even after a crash-recovery. By default RocksDB guarantees this as
+  // long as `WritableFile::Append()` writes are durable. In case the user needs
+  // the guarantee in more situations (e.g., when `WritableFile::Append()`
+  // writes to page cache, but the user desires this guarantee in face of power-
+  // loss crash-recovery), RocksDB offers various mechanisms to additionally
+  // invoke `WritableFile::Sync()` in order to strengthen the guarantee.
+  //
+  // This differs from `kPointInTimeRecovery` in that, in case a corruption is
+  // detected during recovery, this mode will refuse to open the DB. Whereas,
```

Comment:
Just to clarify, `kTolerateCorruptedTailRecords` will refuse to open DB if the corruption is not the trailing record?

Reply:
Yes, and additionally if it's not able to tell if the corruption is the trailing record. For example, an incomplete write to a preallocated file would lead to a checksum mismatch because the record contains a bunch of valid bytes followed by a bunch of zeroes.